src/Carbon/Doctrine/CarbonTzImmutableType.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Carbon\Doctrine;
+
+use Carbon\CarbonImmutable;
+use Carbon\CarbonInterface;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Types\DateTimeTzImmutableType;
+
+class CarbonTzImmutableType extends DateTimeTzImmutableType
+{
+    use CarbonTypeConverter;
+
+    public function getName(): string
+    {
+        return 'carbontz_immutable';
+    }
+
+    public function convertToDatabaseValue($value, AbstractPlatform $platform): ?string
+    {
+        return $this->convertCarbonToDatabaseValue($value, $platform);
+    }
+
+    public function convertToPHPValue($value, AbstractPlatform $platform): ?CarbonInterface
```

Comment:
The type needs to be compatible with the parent method.
```suggestion
    public function convertToPHPValue(mixed $value, AbstractPlatform $platform): ?CarbonInterface
```